src/App.tsx

```diff
@@ -1,27 +1,56 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import 'bulma';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  // 'Saw this method when did studen review'
+  // const handleButtonClick = async (callback: Promise<Good[]>) => {
+  //   setGoods(await callback);
+  // };
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleButtonClick = (func: Promise<Good[]>) => (
```

Comment:
It's not exactly function, it's the result of called function 
```suggestion
  const handleButtonClick = (loadedGoods: Promise<Good[]>) => (
```